testlog/testlog.go

```diff
@@ -31,30 +30,24 @@ func Capture(t *testing.T) func() {
 
 type testLogWriter struct {
 	*testing.T
-	mu      sync.Mutex
+	mu      sync.RWMutex
 	stopped bool
-	ch      chan []byte
 }
 
 func (w *testLogWriter) Write(p []byte) (n int, err error) {
-	w.mu.Lock()
-	defer w.mu.Unlock()
-	if !w.stopped {
-		w.ch <- p
-		return len(p), nil
+	w.mu.RLock()
+	defer w.mu.RUnlock()
+	if w.stopped {
+		// After writer stopped, just log to console
+		p = append([]byte("(logged after test capture stopped) "), p...)
+		_, err := os.Stderr.Write(p)
```

Comment:
Instead of returning an error, write the log message to stderr so that it doesn't get lost.